app/controllers/laptops_controller.rb

```diff
@@ -1,30 +1,22 @@
 class LaptopsController < ApplicationController
-  before_action :set_laptop, only: [:show, :edit, :update, :destroy]
+  before_action :set_asset_directory, only: [:show, :edit, :update, :destroy]
 
-  # GET /laptops
-  # GET /laptops.json
   def index
-    @laptops = Laptop.all
+    @asset_directories = AssetDirectory.where(category: 'Laptops')
   end
 
-  # GET /laptops/1
-  # GET /laptops/1.json
   def show
   end
 
-  # GET /laptops/new
   def new
-    @laptop = Laptop.new
+    @laptop = AssetDirectory.new
```

Comment:
This might be confusing. I think we should check with Ryan if there is a better way to do this. Like maybe one controller that deals with Assets. Maybe you and I need to sit down and think about this as well. Let's make some time to sit together.